adapters/adair-uae.js

```diff
@@ -0,0 +1,309 @@
+  // Fetch both the measurements and meta-data about the locations
+  var tasks = [];
+  _.forEach(stations, function (s) {
+    const sourceName = source.url + s.urlName;
+    var task = function (cb) {
+      request({
+        'rejectUnauthorized': false,
+        'url': sourceName,
+        'method': 'GET'
+      }, function (err, res, body) {
+        if (err || res.statusCode !== 200) {
+          return cb(err || res);
+        }
+        try {
+          const data = JSON.parse(JSON.parse(body).JSONDataResult);
+          cb(null, Object.assign({
+            data
+          }, s));
+        } catch (e) {
+          return cb({message: 'Failure to load data'});
+        }
+      });
+    };
+    tasks.push(task);
+  });
+  async.parallel(tasks, function (err, results) {
+    if (err) {
+      console.log(err);
+      return cb({message: 'Failure to load data urls.'});
+    }
+    // Wrap everything in a try/catch in case something goes wrong
+    try {
+      // Format the data
+      var data = formatData(results);
+      if (data === undefined) {
+        return cb({message: 'Failure to parse data.'});
+      }
+      cb(null, data);
+    } catch (e) {
+      return cb({message: 'Unknown adapter error.'});
+    }
+  });
+};
+
+/**
+ * Given fetched data, turn it into a format our system can use.
+ * @param {array} results Fetched source data and other metadata
+ * @return {object} Parsed and standarized data our system can use
+ */
+var formatData = function (results) {
+  var paramMap = {
+    'PM25': 'pm25',
+    'PM10': 'pm10',
+    'SO2': 'so2',
+    'NO2': 'no2',
+    'CO': 'co',
+    'O3': 'o3'
+  };
+  /**
+   * Given a measurement object, convert to system appropriate times.
+   * @param {object} m A source measurement object
+   * @return {object} An object containing both UTC and local times
+   */
+  var parseDate = function (m) {
+    m = moment(m).format('YYYY-MM-DDHH:mm');
+    var date = moment.tz(m, 'YYYY-MM-DDHH:mm', 'Asia/Dubai');
+    return {utc: date.toDate(), local: date.format()};
+  };
+
+  var measurements = [];
+
+  _.forEach(results, function (s) {
+    var base = {
+      city: s.city,
+      location: s.location,
+      coordinates: s.coordinates,
+      attribution: [
+        {name: 'Environment Agency - Abu Dabhi', url: 'https://www.adairquality.ae/'}
+      ],
+      unit: 'µg/m³'
+    };
+    // Loop over the parameters measured by this station
+    for (var i in s.data) {
+      const date = parseDate(s.data[i].DateTime);
+      for (let [key, value] of Object.entries(s.data[i])) {
+        const param = paramMap[key];
+        if (typeof param !== 'undefined') {
+          var m = _.clone(base);
+          m.parameter = param;
+          m.date = date;
+          m.value = (param === 'co') ? (Number(value) * 1000) : Number(value);
+          measurements.push(m);
+        }
+      }
+    }
+  });
+  return {
+    name: 'unused',
+    measurements: measurements
+  };
+};
+
+const stations = [
```

Comment:
It would be ideal if we could get these programmatically, but I know it's not always possible. Where did the coordinates and city come from? Would be good to at least add a comment with a url

Reply:
The metadata coordinates in the metadata were geometry coordinates, and I could not find any metadata where the exact coordinates were given, so all the coordinates I use here I had to find myself by comparing the map on their site to google maps, and then finding the exact location there

Reply:
Ah got it. Thanks for doing that, it's a lot of work! I'm still a little hesitant to include coordinates claiming 5 decimal point accuracy because they're estimates. But I looked at this url with station metadata again: https://services.arcgis.com/kuR0ZmzEAOg4q3DU/arcgis/rest/services/AirQuality_Pics/FeatureServer/0/query?f=json&where=1%3D1&returnGeometry=true&spatialRel=esriSpatialRelIntersects&outFields= 
And saw that it returns web mercator coordinates (x, y). This [website](https://epsg.io/transform#s_srs=3857&t_srs=4326) can convert those to lat, long decimal coordinates. For example, with that I got `Al Ain Islamic Institute` coordinates to be `24.2190093, 55.7348393` which matches up exactly with the map in the source. It would be great if the coordinates below could be updated using this method and a comment added to explain the method.

Reply:
I also found this ArcGIS [xyToLatLng](https://developers.arcgis.com/javascript/latest/api-reference/esri-geometry-support-webMercatorUtils.html#xyToLngLat) method that could be used to make the conversion programmatically, but from a quick search seems like it will take some work to include and not really worth it.

Reply:
This is perfect, thanks!